File: pubnub-gson/pubnub-gson-docs/src/main/java/com/pubnub/docs/mobilePush/ListChannelsForDeviceOthers.java
package com.pubnub.docs.mobilePush;

import com.pubnub.api.PubNubException;
import com.pubnub.api.enums.PNPushType;
import com.pubnub.api.java.PubNub;
import com.pubnub.docs.SnippetBase;

public class ListChannelsForDeviceOthers extends SnippetBase {
    private void auditPushChannelProvisionsBasic() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/mobile-push#list-channels-for-device-1

        PubNub pubNub = createPubNub();

        // snippet.listChannelsForDevice
        pubNub.auditPushChannelProvisions()
                .deviceId("googleDevice")
                .pushType(PNPushType.FCM)
                .async(result -> { /* check result */ });
        // snippet.end
    }
}

File: pubnub-kotlin/pubnub-kotlin-docs/src/main/kotlin/com/pubnub/docs/mobilePush/ListChannelsForDeviceOthers.kt
package com.pubnub.docs.mobilePush

import com.pubnub.api.enums.PNPushEnvironment
import com.pubnub.api.enums.PNPushType
import com.pubnub.docs.SnippetBase

class ListChannelsForDeviceOthers : SnippetBase() {
    private fun auditPushChannelProvisionsBasic() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/mobile-push#list-channels-for-device-1

        val pubnub = createPubNub()

        // snippet.auditPushChannelProvisions
        // for FCM/GCM
        pubnub.auditPushChannelProvisions(
            pushType = PNPushType.FCM,
            deviceId = "googleDevice"
        ).async { result ->
            result.onFailure { exception ->
                // Handle error
            }.onSuccess { value ->
                // Handle successful method result
            }
        }

        // for APNS2
        pubnub.auditPushChannelProvisions(
            pushType = PNPushType.APNS2,
            deviceId = "appleDevice",
            topic = "myapptopic",
            environment = PNPushEnvironment.DEVELOPMENT
        ).async { result ->
            result.onFailure { exception ->
                // Handle error
            }.onSuccess { value ->
                // Handle successful method result
            }
        }
        // snippet.end
    }
}
